guest/rust/api/src/audio.rs

```diff
@@ -23,23 +23,42 @@ pub struct AudioTrack {
 
 impl AudioTrack {
     /// Set whether or not the track should loop.
-    pub fn looping(&self, looping: bool) -> Self {
-        Self {
-            looping,
-            ..self.clone()
-        }
+    pub fn looping(&mut self, looping: bool) -> &mut Self {
+        self.looping = looping;
+        self
     }
 
     /// Set the volume of the track.
-    pub fn volume(&self, volume: f32) -> Self {
-        Self {
-            volume,
-            ..self.clone()
-        }
+    pub fn volume(&mut self, volume: f32) -> &mut Self {
```

Comment:
Should the volume be clamped from [0,1)?

In some cases you want to make the source *louder* than normal if it is very quiet, I've had to do that for previous effects.

Also, what is the difference between `gain` `volume` and `amp`?

Reply:
> Should the volume be clamped from [0,1)?
> 
> In some cases you want to make the source _louder_ than normal if it is very quiet, I've had to do that for previous effects.
> 
> Also, what is the difference between `gain` `volume` and `amp`?

Good point. Need to unify the usage here. I actually had another term `scale` 😂

Reply:
I have removed the clamp. For the internal API, I suggest that we keep it as is. The users should only need to know and use `volume` at the moment.